### DISCUSSION TITLE
Having trouble uploading via serial port

### DISCUSSION THREAD

**Discussion by noweare1** [SpenceKonde/ATTinyCore#660] [General]
Hello,
I am using an ATTiny85 dip 8 pin package. I am able to download sketches via the USBasp programmer and burn the 
bootloader (Chip: ATTiny85(Optiboot) ) 

Disconnecting the USBasp programmer and connecting the FTDI then selecting sketch->upload 

I get the error:
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x00
which continues until attempt 10 is reached.

I did addthe 0.1 cap between DTR and Reset (PB5)  and a 10K resistor between Reset and vcc.
I have a scope so if there is a signal that I need to verify then I can do that.

Thank you

**Comment by pcfreak1201** [comment]
So you use the bare chip on a breadboard? Did you follow " Spences' " requirements?
[Minimal requirements ](https://github.com/SpenceKonde/ATTinyCore/blob/master/Wiring.md)
May be you didn't cross RX and TX? Could you send a photo?

**Reply by noweare1** [reply]
Yes, I followed all the wiring requirements. I already double checked them. Will check them again. Everything is on
a breadboard, RX, TX are in the right place. Here is the schematic of what I have.
![ATTiny85_program_ckt](https://user-images.githubusercontent.com/22323751/152646251-392f8920-c03b-43fb-b37f-6a92aacbd167.png)

**Comment by pcfreak1201** [comment]
Well, so we can debate - of course we are in discussions. :-) But try this:

> Diode (any) between Reset and Vcc, band towards Vcc.

That's what me helped with this problem...

**Reply by noweare1** [reply]
So I found that I wired a pin incorrectly which caused the attiny85 to prematurely retire.
I replaced the attiny85 and was able to download via ftdi board. I  had the diode in the circuit 
at this time. I removed the diode to see if it would stop working but it continued to work 
without the diode.  So the mystery continues. Sorry I could not confirm  or deny the usefulness
of the diode.

**Reply by pcfreak1201** [reply]
As I said :-) It depends if there is already a diode e.g. for protection on the USB-serial-converter or other kind of impendance changer...
As long as it works - all fine!

**Comment by SpenceKonde** [comment]
The diode generally speaking is voodoo. 

It'sthere because some "clever" people on the forums way back when obvserved people who reported that the part would not upload, despite having a well-behaved bootloader on it that should be unbrickable. But they repotted that "burn bootloader" fixed the problem. These folks concluded that the brief pulse when reset is released just *must* be putting it into HV programming mode, and ambient noise and voltages on the pins just happened to match a programming command, and one that could corrupt the bootloader. 

This is lottery winning chances. I have observed the same pattern in support requests, but I observe a few things in addition:
Nobody has ever dumpedthe flash of a chip, which would provide proof of bootloader corruption
All thepeople reporting this are inexperienced 
The errors they reported getting were very generic. 

My interpretation was that disconnecting everthing, reconnecting an isp programmer, then disconnecting and reconnecting everything was what fixed the problem, everything to do with poor wiring and nothing to do with bootloaders.

**Comment by SpenceKonde** [comment]
Now, on a modern tinyAVR, yes that diode is essential if you have the pin fused as reset, because that voltage spike could be enough to  do a HV override of the pin function and turn it back into a UPDI pin. 

On other modern AVR, it's unclear. But on classic AVRs, it's just voodoo

**Reply by pcfreak1201** [reply]
Well, I had the problem with my "bare metal UNO breakout" and haven't since soldering in the diode - so I'm not spiritual and believe you ;-)

On the other hand, had I trouble with SerialUPDI on a cheap CH340, but never investigated it. So the info with the HV-peak is very interesting at all.

But because it's already solved, just my 2 cents ... crossing fingers for the new V2.0.0. Looking forward to it ((-:

**Reply by SpenceKonde** [reply]
It most certainly has been challenging to nail down a consistently working SerialUPDI design which works for all people across all adapters and platforms - and there's never been a smoking gun regarding what actually caused an given problem, but the new writedelay options in 1.4.7 should help if it was megaTinyCore specific.

**Reply by SpenceKonde** [reply]
And by the way, 2.0.0 i availabe for public testing via manual installation.
